Aim of this notebook:
-exploring datasets of socio demographic informations

In [5]:
#Importer les packages 
import pandas as pd
from pathlib import Path

In [ ]:
from pathlib import Path



AttributeError: type object 'set' has no attribute 'cwd'

In [22]:
#indiquer le répertoire des données
#donnees="C:/Users/user\Documents\Municipales_2026\Données"
#lire les données
#age 
#age=pd.read_csv("C:/Users/user/Documents/Municipales_2026/Données/Age_csp/Age_csp/agesexcommunes.csv")
#age=pd.read_csv(Path("Données/Age_csp/Age_csp/agesexcommunes.csv"))
age = pd.read_csv(
    Path("..") / "Données" / "Age_csp" / "Age_csp" / "agesexcommunes.csv"
)


C:\Users\user\AppData\Local\Temp\ipykernel_5312\252573482.py:7: DtypeWarning: Columns (0,2) have mixed types. Specify dtype option on import or set low_memory=False.
  age = pd.read_csv(


In [ ]:

age


dep           nomdep codecommune              nomcommune  poph0141962  \
0      25            DOUBS       25443               PALANTINE          5.0   
1      70      HAUTE-SAONE       70440               RECOLOGNE          3.0   
2      21        COTE-D'OR       21276   FONTAINES-EN-DUESMOIS         32.0   
3      31    HAUTE-GARONNE       31266                  LAHAGE          6.0   
4       6  ALPES-MARITIMES        6076                 LIEUCHE          1.0   
...    ..              ...         ...                     ...          ...   
37980  23           CREUSE       23231  SAINT-PIERRE-DE-FURSAC         85.0   
37981  52      HAUTE-MARNE       52238              HARRICOURT          8.0   
37982  48           LOZERE       48049                  CHIRAC         84.0   
37983  61             ORNE       61205                  HEUGON         50.0   
37984  14         CALVADOS       14470        NORREY-EN-BESSIN         32.0   

       poph15391962  poph40591962  poph60p1962  popf0141962  popf15391962  \
0               5.0           6.0          3.0          7.0           6.0   
1               6.0           4.0          2.0          3.0           5.0   
2              37.0          24.0         16.0         29.0          22.0   
3              94.0          17.0         10.0          8.0          14.0   
4               3.0           4.0          2.0          2.0           5.0   
...             ...           ...          ...          ...           ...   
37980         155.0         131.0        124.0        102.0         133.0   
37981          13.0          10.0          8.0          8.0          10.0   
37982         123.0          79.0         68.0        142.0         112.0   
37983          54.0          45.0         33.0         60.0          57.0   
37984          37.0          21.0         11.0         31.0          25.0   

       ...  perprop0141957  perprop60p1957  perpropf1958  perage1958  \
0      ...        0.929899        0.093060      0.946069    0.006167   
1      ...        0.355285        0.711124      0.112636    0.088098   
2      ...        0.938310        0.188820      0.043178    0.144777   
3      ...        0.003273        0.089764      0.001752    0.149059   
4      ...        0.069635        0.323348      0.509821    0.861277   
...    ...             ...             ...           ...         ...   
37980  ...        0.085405        0.952224      0.734750    0.980018   
37981  ...        0.147660        0.659091      0.464339    0.898641   
37982  ...        0.783682        0.893864      0.973157    0.652919   
37983  ...        0.833193        0.591287      0.493604    0.358725   
37984  ...        0.718533        0.267685      0.743903    0.446151   

       perprop0141958  perprop60p1958  perpropf1959  perage1959  \
0            0.929899        0.093060      0.946069    0.006167   
1            0.355285        0.711124      0.112636    0.088098   
2            0.938310        0.188820      0.043178    0.144777   
3            0.003273        0.089764      0.001752    0.149059   
4            0.069635        0.323348      0.509821    0.861277   
...               ...             ...           ...         ...   
37980        0.085405        0.952224      0.734750    0.980018   
37981        0.147660        0.659091      0.464339    0.898641   
37982        0.783682        0.893864      0.973157    0.652919   
37983        0.833193        0.591287      0.493604    0.358725   
37984        0.718533        0.267685      0.743903    0.446151   

       perprop0141959  perprop60p1959  
0            0.929899        0.093060  
1            0.355285        0.711124  
2            0.938310        0.188820  
3            0.003273        0.089764  
4            0.069635        0.323348  
...               ...             ...  
37980        0.085405        0.952224  
37981        0.147660        0.659091  
37982        0.783682        0.893864  
37983        0.833193        0.591287  
37984        0.718533 

In [ ]:
list(age.columns)

['dep',
 'nomdep',
 'codecommune',
 'nomcommune',
 'poph0141962',
 'poph15391962',
 'poph40591962',
 'poph60p1962',
 'popf0141962',
 'popf15391962',
 'popf40591962',
 'popf60p1962',
 'ageh1962',
 'agef1962',
 'poph0141968',
 'popf0141968',
 'poph15391968',
 'popf15391968',
 'poph40591968',
 'popf40591968',
 'poph60p1968',
 'popf60p1968',
 'ageh1968',
 'agef1968',
 'poph0141975',
 'popf0141975',
 'poph15391975',
 'popf15391975',
 'poph40591975',
 'popf40591975',
 'poph60p1975',
 'popf60p1975',
 'ageh1975',
 'agef1975',
 'poph0141982',
 'popf0141982',
 'poph15391982',
 'popf15391982',
 'poph40591982',
 'popf40591982',
 'poph60p1982',
 'popf60p1982',
 'ageh1982',
 'agef1982',
 'poph0141990',
 'popf0141990',
 'poph15391990',
 'popf15391990',
 'poph40591990',
 'popf40591990',
 'poph60p1990',
 'popf60p1990',
 'ageh1990',
 'agef1990',
 'poph0141999',
 'popf0141999',
 'poph15391999',
 'popf15391999',
 'poph40591999',
 'popf40591999',
 'poph60p1999',
 'popf60p1999',
 'ageh1999',
 'agef1999',
 '

In [ ]:
#regarder ce que sont ces colonnes (age+ date) 
age.ageh2022 #age moyen dans la commune ?? pas sûre 

0        18.000000
1        63.200001
2        44.736843
3        47.094112
4        45.500000
           ...    
37980    18.000000
37981    18.000000
37982    18.000000
37983    18.000000
37984    18.000000
Name: ageh2022, Length: 37985, dtype: float64

noms : 
poph =pop homme
popf= pop femme 

CCL : 
pour chaque année découpage pop H et F pour 0-14, 15-39, 40-59, 60 et + 
age moyen pour F H et Total ( pas sûr)
perage + date ?????? 

37985 communes × 2385 variables